sympy/matrices/common.py

```diff
@@ -2540,3 +2554,85 @@ def classof(A, B):
             return A.__class__
 
     raise TypeError("Incompatible classes %s, %s" % (A.__class__, B.__class__))
+
+
+def sumprodsimp(a, b, simplify=True):
+    """Sum-of-products with optional intermediate product simplification
```

Comment:
Really this function is computing a dot-product. Maybe that makes more sense as a name

Reply:
Looking at the places this function is called I wonder whether actually a sumsimp taking a single iterable would be better than dotprodsimp taking two iterables.